packages/oidc-middleware/test/e2e/harness/server.js

```diff
@@ -3,45 +3,74 @@ const session = require('express-session');
 const uuid = require('uuid');
 const constants = require('../util/constants');
 const { ExpressOIDC } = require('../../../index.js');
+const path = require('path');
 
-const app = express();
-
-app.use(session({
-  secret: uuid(), // this will invalidate all sessions on each restart
-  resave: true,
-  saveUninitialized: false
-}));
+module.exports = class DemoServer {
+  constructor(oidcOptions) {
+    this.oidcOptions = oidcOptions;
+  }
+  start() {
+    const app = express();
+    this.app = app;
+    app.use(session({
+      secret: uuid(), // this will invalidate all sessions on each restart
+      resave: true,
+      saveUninitialized: false
+    }));
 
-const oidc = new ExpressOIDC({
-  issuer: constants.ISSUER,
-  client_id: constants.CLIENT_ID,
-  client_secret: constants.CLIENT_SECRET,
-  redirect_uri: constants.REDIRECT_URI
-});
+    app.set('view engine', 'ejs');
+    app.set('views', path.join(__dirname, 'views'));
 
-app.use(oidc.router);
+    const oidc = new ExpressOIDC(this.oidcOptions);
+    app.use(oidc.router);
 
-app.get('/', (req, res) => {
-  if (req.userinfo) {
-    res.send(`Hello ${req.userinfo.sub}! Welcome home`);
-  } else {
-    res.send('Hello World!');
-  }
-});
+    app.get('/', (req, res) => {
+      if (req.userinfo) {
+        res.send(`Hello ${req.userinfo.sub}! Welcome home`);
+      } else {
+        res.send('Hello World!');
+      }
+    });
 
-app.get('/protected', oidc.ensureAuthenticated(), (req, res) => {
-  res.send(JSON.stringify(req.userinfo));
-});
+    app.get('/protected', oidc.ensureAuthenticated(), (req, res) => {
+      res.send(JSON.stringify(req.userinfo));
+    });
 
-app.get('/logout', (req, res) => {
-  req.logout();
-  res.redirect('/');
-});
+    app.get('/logout', (req, res) => {
+      req.logout();
+      res.redirect('/');
+    });
 
-oidc.on('ready', () => {
-  app.listen(constants.PORT, () => console.log(`Test app listening on port ${constants.PORT}!`));
-});
+    return new Promise((resolve, reject) => {
+      oidc.on('error', err => {
+        console.log('Unable to start the server', err);
+        reject(err);
+      });
+      oidc.on('ready', () => {
+        const enableDestroy = require('server-destroy');
```

Comment:
what is this?

Reply:
That library will kill all connections to the server, so that we can un-bind it from the port.  This allows us to use the same process to manage multiple express apps.